packages/cli/src/TestWebhooks.ts

```diff
@@ -219,16 +221,18 @@ export class TestWebhooks implements IWebhookManager {
 			webhook.isTest = true;
 
 			/**
-			 * Remove additional data from webhook because:
-			 *
-			 * - It is not needed for the test webhook to be executed.
-			 * - It contains circular refs that cannot be cached.
+			 * Additional data cannot be cached because of circular refs.
+			 * Hence store the `userId` and recreate additional data when needed.
 			 */
 			const { workflowExecuteAdditionalData: _, ...rest } = webhook;
 
+			rest.userId = userId;
+
 			try {
 				await workflow.createWebhookIfNotExists(webhook, NodeExecuteFunctions, 'manual', 'manual');
 
+				workflowEntity.staticData = workflow.staticData;
```

Comment:
I have a slight concern here that this might break the behavior for production executions.

As per your comment, static data contains information such as `{ 'node:Linear Trigger': { webhookId: '6265f69f-8937-40b5-8740-edd74abc0f18' }}`.

if the workflow is currently active (i.e. production webhook registered), the static data would contain the webhook ID used in production.

Now while the workflow is active, I am working on the workflow and make changes ot it, and click to manually run to test. This would override the actual production webhook ID.

Now we shifted the problem to another place. The production webhook will never deregister since we lost the ID.

Reply:
Please see [`23e5743`](https://github.com/n8n-io/n8n/pull/8247/commits/23e574341e5d1fbfbfd9a5030bb52b91fa845709)